### ISSUE TITLE
ofAVFoundationVideoPlayer async loading problems

### ISSUE THREAD

**Issue by HalfdanJ** [openframeworks/openFrameworks#3628]
I'm not exactly sure what the intended behavior is, i can just see that something has changed with the new ofAVFoundationVideoPlayer on OS X #3590, take the following code:

```{.cpp}
void ofApp::setup(){
    fingerMovie.loadMovie("movies/fingers.mov");    
    ofLog()<<fingerMovie.getWidth();
}
```
This printed `320` in the old video player, but because the new vide player loads the video async, this now prints `0` until it laters has been loaded. I can see a huge benefit in having it load it async for other cases, but i think there should either be a parameter for specifying it, a global flag for it or having an event I can listen to (as far as i can see there is none right now). 

@julapy

**Comment by julapy** [comment]
yep i agree.

we can add a function to `ofBaseVideoPlayer`,
`setAsyncLoad(bool value)`
which would need to be called before calling `load(string name)`
by default asynchronous load should be turned off as that is the expected behaviour.

alternatively we could have two different load functions.
`load(string name)`
`loadAsync(string name)`

i personally would prefer the first approach, setting a load async flag before loading.

cc @openframeworks/core @openframeworks/video

**Comment by arturoc** [comment]
just to add that the gstreamer video player already works like it's suggested here. by default it loads synchronously and then it has a flag `setAsynchronousLoad` which makes it load asynchronously. the `loadAsync` method is even a better idea in my opinion. so +1 to add this to ofBaseVideoPlayer and ofVideoPlayer

**Comment by mattfelsen** [comment]
Here's a comment/sample implementation for AVFoundation sync load/update from another thread. Looks like it's was done against the CLOUDS-based player so it may need some modification to get working with the *new* new player switch, but just linking to it for discussion/possible quick fix for Jonas :)

https://github.com/openframeworks/openFrameworks/pull/3200#issuecomment-70400878

**Comment by HalfdanJ** [comment]
@mattfelsen I actually just did an implementation (in a different way that seems to work)

**Comment by mattfelsen** [comment]
Haha @HalfdanJ, yea your PR popped up literally as I was clicking to submit my comment ;)

**Comment by bilderbuchi** [comment]
closed by #3653, I think
